assets/js/base/utils/tracks.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
+function isTracksAvailable() {
+	return typeof window.wcTracks.recordEvent === 'function';
+}
+
+export const recordEditorEvent = function( event, props ) {
+	if ( ! isTracksAvailable() ) {
+		return;
+	}
+
+	// Force prefix - our editor events will be 'wcadmin_blocks_*'.
+	const blocksPrefix = 'blocks_';
+
+	const postData = select( 'core/editor' );
+	window.wcTracks.recordEvent( blocksPrefix + event, {
+		// Automatically include post id and post type props.
+		post_id: postData.getCurrentPostId(),
+		post_type: postData.getCurrentPostType(),
```

Comment:
I know this are examples, but whats the purpose in tracking Post ID and type? Is this the type of usage data we actually need? I thought we'd be more interested in how blocks themselves are interacted with. Including editor specific selects also limits future usage right?

Reply:
This is thinking ahead to an important event we want to track – when the merchant adds the cart or checkout to a page. One interesting prop that we'd like to track is whether this is the woo cart page (vs. a post, or a "test" page). 

These props are initial work towards this:

- We want to confirm that the block is added to a page (as opposed to blog post or other type).
- If we have the post ID, we can compare this to the ID for the cart page, and then we can add a prop for if merchant is editing the real cart page, which will help us understand how the blocks are being used – are merchants using cart block for real, or experimenting with it on a test page?

Having said that, the details of these standard props are something that we should iterate on as we add events. For example, `post_id` may not be that useful on its own, possibly post slug might be more useful.